src/app-layer-parser.c

```diff
@@ -935,6 +935,8 @@ void AppLayerParserTransactionsCleanup(Flow *f, const uint8_t pkt_dir)
     SCLogDebug("start min %"PRIu64, min);
     bool skipped = false;
     // const bool support_files = AppLayerParserSupportsFiles(f->proto, f->alproto);
+    uint64_t tofree_stack[total_txs];
```

Comment:
this can be a large allocation? 8 bytes * total_txs, where total txs is 1024 for some protocols? So 8k. But is this even a count, or is it just the highest tx_id?

Reply:
return of `AppLayerParserGetTxCnt`

Reply:
so, what is the max stack allocation size?

Reply:
I am afraid it can get too big, there is no generic max tx for all protocols...